models/models.js

```diff
@@ -1,65 +1,132 @@
-         }
 
-        return result.rows[0];
+        if(result.rows.length === 0) {
+            return Promise.reject({ status: 404, message: 'Username does not exist'});
+        }
+      return result.rows[0];
     })
-}
+};
+module.exports.checkUsernameExists = checkUsernameExists;
+
+exports.fetchArticleById = (id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1;`, [id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({ status: 404, message: "item does not exist" });
+      }
+
+      return result.rows[0];
+    });
+};
 
 //5
 exports.fetchArticles = () => {
-    const articlesQuery = `
+  const articlesQuery = `
     SELECT a.article_id, a.title, a.author, a.topic, a.created_at, a.votes, a.article_img_url, COUNT(c.comment_id) AS comment_count 
 FROM articles AS a 
 LEFT JOIN comments AS c 
 ON c.article_id = a.article_id
 GROUP BY a.article_id
 ORDER BY a.created_at DESC; 
 `;
-    return db.query(articlesQuery)
-        .then(({rows}) => {
-           return rows
-    })
-}
+  return db.query(articlesQuery).then(({ rows }) => {
+    return rows;
+  });
+};
 //6
 exports.selectArticleComments = (articleId) => {
-    return db.query(`
+  return db
+    .query(
+      `
     SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC; 
-    `,  [articleId])
+    `,
+      [articleId]
+    )
     .then((result) => {
-        return result.rows;
+      return result.rows;
+    });
+};
+//7
+exports.insertComment = (newComment, id) => {
+  const { username, body } = newComment;
+
+  return Promise.all([
+    checkUsernameExists(newComment.username),
+    checkArticleExists(id)
+  ])
+    .then(([usernameExists, articleExists]) => {
+      if (!usernameExists) {
+        return Promise.reject({ status: 404, message:'Username does not exist' });
+      }
+      if (!articleExists) {
+        return Promise.reject({ status: 404, message: "Article does not exist" });
+      }
+    
+    return db.query(
+      "INSERT INTO comments (author, body, article_id) VALUES ($1, $2, $3 ) RETURNING *;",
+      [username, body, id] );
     })
-}
-   
+    
+    .then((result) => {
+      return result.rows[0];
+    });
+};
+
+//8
+exports.updateArticle = async (article_id, inc_votes) => {
+
+  const articleExistsResult = await db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id]);
+  if (articleExistsResult.rows.length === 0) {
+    return Promise.reject({ status: 404, message: 'Article does not exist' });
+  }
 
+  //check current votes and compare to the increment, if currentVotes - inc_votes < 0 => bad request
+  const currentVotesResult = await db.query(`SELECT votes FROM articles WHERE article_id = $1;`, [article_id])
+  const currentVotes = currentVotesResult.rows[0].votes;
  
+
+  if ((currentVotes + inc_votes) < 0) {
+    return Promise.reject({ status: 400, message: 'Votes cannot be negative' })
+  }
```

Comment:
As I said earlier I think votes should be allowed to be negative, to allow "unvoting"